### PR TITLE
misc: minor updates in tests, readme, dependabot and release-drafter

### PR DESCRIPTION
**Pull request by georgebisbas**

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -3,32 +3,39 @@
 
 name: Python application
 
-on: [push, pull_request]
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the master branch
+  push:
+    branches:
+      - main
```
Comment (webmiche) [inline .github/workflows/python-app.yml:11]: I am not sure we want to just run this on pushes to main, I feel there is value in having a CI ran for each push to development branches. And the jobs are fairly light-weight, so we are not wasting a lot of computation.

File: requirements-optional.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+yapf<0.33.*
```
Comment (webmiche) [inline requirements-optional.txt:1]: Maybe mention the splitup of requirements and optional requirements in the README?

**Review by webmiche — APPROVED** [review]
Thanks a lot for this PR, I think this makes the project a step closer to more structured development compared to our more researchy aproach atm.

File: requirements-optional.txt (hunk 2 above)
Comment (georgebisbas) [reply requirements-optional.txt:1]: good idea

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    schedule:
+      interval: "daily"
+    # Add assignees
+    assignees:
+      - "math-fehr"
+      - "georgebisbas"
+    commit-message:
+      prefix: "pip prod"
+      prefix-development: "pip dev"
+      include: "scope"
```
Comment (math-fehr) [inline .github/dependabot.yml:19]: Could we add one newline here?

File: .github/release-drafter.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    label: 'CI'
+  - title: 'Installation'
+    labels:
+      - 'docker'
+      - 'installation'
+      - 'dependencies'
+change-template: '- $TITLE @$AUTHOR (#$NUMBER)'
+template: |
+  ## Changes
+  $CHANGES
```
Comment (math-fehr) [inline .github/release-drafter.yml:42]: Can we add a newline here as well?

File: .github/release-drafter.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+categories:
+  - title: 'API'
+    label: 'API'
+  - title: 'Examples'
+    label: 'examples'
+  - title: 'Documentation'
+    label: 'documentation'
+  - title: 'Compiler'
+    label: 'compiler'
+  - title: 'MPI'
```
Comment (math-fehr) [inline .github/release-drafter.yml:13]: Do we need MPI, GPU, arch, bug-C/performance, benchmarking categories?
I feel that those are specific to Devito, and do not make much sense for xDSL?

File: .github/workflows/yapf-check.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install formatting requirements
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements-optional.txt
+    - name: Check format with yapf 
+      run: |
+        # Returns en error code if the files are not formatted
+        yapf --diff -r .
```
Comment (math-fehr) [inline .github/workflows/yapf-check.yml:37]: Could you add a newline here?

File: .gitignore [hunk 7]
```diff
@@ -140,3 +140,7 @@ cython_debug/
 # PyCharm IDE
 *.idea/
 .vscode/settings.json
+
+
+# lit tests results
+tests/filecheck/.lit_test_times.txt
```
Comment (math-fehr) [inline .gitignore:146]: Same here, could you add a newline?

**Review by math-fehr — APPROVED** [review]
Thanks a lot for making the package much cleaner!
I just left some nit comments

File: .github/dependabot.yml (hunk 3 above)
Comment (georgebisbas) [reply .github/dependabot.yml:19]: Yes

File: .github/release-drafter.yml (hunk 4 above)
Comment (georgebisbas) [reply .github/release-drafter.yml:42]: Yes

File: .github/release-drafter.yml (hunk 5 above)
Comment (georgebisbas) [reply .github/release-drafter.yml:13]: Definitely agree, this was just a sketch

File: .github/workflows/yapf-check.yml (hunk 6 above)
Comment (georgebisbas) [reply .github/workflows/yapf-check.yml:37]: Yes

File: .gitignore (hunk 7 above)
Comment (georgebisbas) [reply .gitignore:146]: Yes

**Comment by math-fehr** [comment]
Sorry for the delay, this looks good to me now!
Can I squash and merge it, or do you want to add anything else to it?

**Pull request merged**